lib/pure/algorithm.nim

```diff
@@ -93,6 +96,13 @@ proc fill*[T](a: var openArray[T], value: T) =
     assert a == [4, 4, 4, 4, 4, 4]
   fillImpl(a, 0, a.high, value)
 
+template fillWith*[T](a: var openArray[T], f: untyped) =
+  ## Fills the container ``a`` with ``f``.
```

Comment:
The difference between ``fill`` and ``fillWith`` is unclear, you need to improve the docs.

Reply:
Documentation updated with more illustrative example which shows the difference between `fill` and `fillWith`.